### ISSUE TITLE
RPI4 not compatible with rpi compiler config

### ISSUE THREAD

**Issue by RSDNTWK** [HaxeFoundation/hxcpp#959]
The Raspberry Pi compiler strings are not compatible with RPI4 since it uses mesa drm not VC4 and also doesn't work with arm64.

**Comment by VMGuy23** [comment]
This is quite a big thing

The only Pi with great performance Pi 4.

**Comment by RSDNTWK** [comment]
Compiling for the older Raspberry Pi's do work however there are massive performance issues.

**Comment by mobilegmYT** [comment]
I think pi4 might work at like 10-20 fps with the right optimisations.

**Comment by RSDNTWK** [comment]
Probably more if overclocked.

**Comment by VMGuy23** [comment]
I dont think it supports Pi 4 processor

**Comment by VMGuy23** [comment]
> Probably more if overclocked.

It can go up to 2.16 Ghz

**Comment by mobilegmYT** [comment]
The hardware limit is 2.5, I've overclocked all the way to 2.35

**Comment by VMGuy23** [comment]
Nice
Is GNOME still under 10 fps

**Comment by RSDNTWK** [comment]
Pixel or xfce doesn't lag as much.

**Comment by mobilegmYT** [comment]
I mean currently, using the RPI3 preset to build something on RPi4 works quite well too. I tried building the sample BunnyMark demo with lime, and even building with the rpi flag meant for pi 3 on a pi 4, it can render 3490 bunnies bouncing around at a decent 32 FPS.

**Comment by RSDNTWK** [comment]
Sadly when we tried to compile anything on the pi4 the program just crashes trying to start.

**Comment by mobilegmYT** [comment]
> Sadly when we tried to compile anything on the pi4 the program just crashes trying to start. This was due to it trying to initialise the vc4 not kmsdrm.

For me, all the sample lime games run fine but Friday Night Funkin had a compile error. I was using @gepatto's lime fork which should have the SDL/kmsdrm stuff in it, but it still didn't fix the compile issue.

**Comment by RSDNTWK** [comment]
We were able to get fnf by using the git version of hxcpp to compile the ndll file for rpi. It just refused to start on pi4 but worked on pi3 just fine but was slow.

**Comment by gepatto** [comment]
I don't think these are hxcpp issues.
I am using hxcpp 4.2.1 from haxelib on my Pi4, not a git version.
mesa drm is not part of haxe or hxcpp,  It's used in SDL. 

As @mobilegmYT says you can try my experimental lime fork which uses a dev-branch-fork of SDL.
You'll need the proper dependencies to get things running on a Pi4.
But you will **have to** optimize a more complex application for the Pi

And I only tested this on a regular RaspberryPi OS ( not 64bit  or other linux distros )

**Comment by VMGuy23** [comment]
I'm talking about `lime build rpi` on a regular machine. Result not work on Pi 4

**Comment by mobilegmYT** [comment]
> I don't think these are hxcpp issues.
> I am using hxcpp 4.2.1 from haxelib on my Pi4, not a git version.
> mesa drm is not part of haxe or hxcpp, It's used in SDL.
> 
> As @mobilegmYT says you can try my experimental lime fork which uses a dev-branch-fork of SDL.
> You'll need the proper dependencies to get things running on a Pi4.
> But you will **have to** optimize a more complex application for the Pi
> 
> And I only tested this on a regular RaspberryPi OS ( not 64bit or other linux distros )

What depends? I got all the apt depends on the lime gh page except for multilib since their not in apt. Maybe you could try building a armhf Friday Night Funkin build?

**Comment by gepatto** [comment]
> I'm talking about `lime build rpi` on a regular machine. Result not work on Pi 4

That's right, you would need to setup cross compiling tools on win/linux/mac to be able to compile raspberry pi.
I never tried that but I know it's possible. 

I compile everything on the pi itself .

**Comment by gepatto** [comment]
> What depends? I got all the apt depends on the lime gh page except for multilib since their not in apt. Maybe you could try building a armhf Friday Night Funkin build? That would be awesome.

I meant mesa dependencies.

I don't mean to be blunt but I don't think the hxcpp issue tracker is the place to discuss this (and fnf) further but you can find me on the haxe and/or openfl discord server where I can give more details on pi4 support
